llvm/test/Transforms/LoopVectorize/AArch64/vector-call-linear-args.ll

```diff
@@ -0,0 +1,275 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 2
+; RUN: opt < %s -passes=loop-vectorize,instsimplify -force-vector-interleave=1 -S | FileCheck %s
```

Comment:
Hi @huntergr-arm ,
wouldn't be better to remove the "test_linear_with_mask" test function, but:
* have more than 1 mapping available per scalar call, (as that is most likely going to be real life scenario), so we can test if LV is choosing the right one,
* add more RUN lines?
1. RUN: opt < %s -passes=loop-vectorize,instsimplify -force-vector-interleave=1 //only Neon mappings used
2.RUN: opt < %s -passes=loop-vectorize,instsimplify -force-vector-interleave=1 -mattr=+sve //only sve mappings used
3.RUN: opt < %s -passes=loop-vectorize,instsimplify -force-vector-interleave=1 -mattr=+sve -prefer-predicate-over-epilogue=predicate-dont-vectorize //sve mappings used, but also Neon masked mapping used when available

Reply:
I also think we should add at least one test case where step is negative, as that is allowed I think. 
The VFABI mangling allows it, from the specification: "For example, linear(val(i):-3) gives Ln3 when the type of i is integer"